src/unittest/python/test_face_verify.py

```diff
@@ -0,0 +1,152 @@
+"""
+----------------------------------------------------------------------
+Authors: Stephan Nell
+----------------------------------------------------------------------
+Unit tests for the Face Verify
+All the images used are from public domain and are copyright free
+----------------------------------------------------------------------
+"""
+import pytest
+import cv2
+import os
+from verification.face_verify import FaceVerify
+from image_preprocessing.blur_manager import BlurManager
+
+# Constants path to trained data for Shape Predictor.
+SHAPE_PREDICTOR_PATH = "{base}/../../main/python/image_preprocessing/trained_data/shape_predictor_face_landmarks.dat".format(
+    base=os.path.abspath(os.path.dirname(__file__)))
+
+FACE_RECOGNITION_PATH = "{base}/../../main/python/image_preprocessing/trained_data/dlib_face_recognition_resnet_model_v1.dat".format(
+    base=os.path.abspath(os.path.dirname(__file__)))
+
+TEMPLATE_DIR = "{base_path}/../../main/python/image_preprocessing/templates/".format(
+    base_path=os.path.abspath(os.path.dirname(__file__)))
+
+test_image_colour = cv2.imread(TEMPLATE_DIR + "temp_flag.jpg")
+
+thanks_obama = cv2.imread(TEMPLATE_DIR + "obama.jpg")
+
+# Soap Joe official example for Fraud Detection for South African ID books
+soap_joe = cv2.imread(TEMPLATE_DIR + "soapJoeExample.jpg")
+
+
+def test_face_verify_constructor():
+    """
+    Test to see if constructor without path passed
+    """
+    with pytest.raises(TypeError):
+        FaceVerify()
+
+
+def test_face_verify_constructor_2():
+    """
+    Test to see if constructor with one path passed and other omitted
+    """
+    with pytest.raises(TypeError):
+        FaceVerify(SHAPE_PREDICTOR_PATH)
+
+
+def test_face_verify_constructor_3():
+    """
+    Test to see if constructor one path and one incorrect type
+    """
+    with pytest.raises(TypeError):
+        FaceVerify(SHAPE_PREDICTOR_PATH, 1)
+
+
+def test_face_verify_constructor_4():
+    """
+    Test to see if constructor one path and one incorrect type
+    """
+    with pytest.raises(TypeError):
+        FaceVerify(1, FACE_RECOGNITION_PATH)
+
+
+def test_face_verify_constructor_5():
+    """
+    Test to see if constructor both paths incorrect
+    """
+    with pytest.raises(TypeError):
+        FaceVerify(1, FACE_RECOGNITION_PATH)
+
+
+def test_face_verify_constructor_6():
+    """
+    Test to see if constructor both paths correct
+    """
+    FaceVerify(SHAPE_PREDICTOR_PATH, FACE_RECOGNITION_PATH)
```

Comment:
Test whether these values are actually assigned correctly to the correct attributes.

Reply:
I was waiting for the OpenCV fix before asserting some of these values. Accuracy was wildly inaccurate with these low-quality images. I will finally be able to test these now at least.